src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -17,216 +17,113 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { D3DragEvent, drag as d3Drag } from 'd3-drag'
 import { easeCubic } from 'd3-ease'
-import { Simulation } from 'd3-force'
-import { BaseType, select as d3Select } from 'd3-selection'
-import { Transition } from 'd3-transition'
-import { D3ZoomEvent, zoom as d3Zoom } from 'd3-zoom'
-
-import * as vizRenderers from '../renders/init'
+import { BaseType, Selection, select as d3Select } from 'd3-selection'
+import { D3ZoomEvent, ZoomBehavior, zoom as d3Zoom } from 'd3-zoom'
+
+import { ZOOM_SCALE_EXTENT } from '../constants'
+import {
+  node as nodeRenderer,
+  relationship as relationshipRenderer
+} from '../renders/init'
 import { nodeMenuRenderer } from '../renders/menu'
+import { ForceSimulation } from './ForceSimulation'
 import Graph from './Graph'
 import GraphGeometry from './GraphGeometry'
 import Relationship from './Relationship'
 import VizNode from './VizNode'
-import { Layout } from './layout'
+import {
+  nodeEventHandlers,
+  relationshipEventHandlers,
+  zoomEventHandler
+} from './mouseEventHandlers'
 import GraphStyle from 'browser/modules/D3Visualization/graphStyle'
 
 export type MeasureSizeFn = () => { width: number; height: number }
-export type VizObj = {
-  style: any
-  trigger: (_event: any, ..._args: any[]) => void
-  zoomInClick: (el: any) => { zoomInLimit: boolean; zoomOutLimit: boolean }
-  zoomOutClick: (el: any) => { zoomInLimit: boolean; zoomOutLimit: boolean }
-  boundingBox: () => void
-  resize: () => void
-  update: (options: {
-    updateNodes: boolean
-    updateRelationships: boolean
-    precompute?: boolean
-  }) => void
-}
 
-const noOp = () => undefined
-const vizFn = function (
-  el: SVGElement,
-  measureSize: MeasureSizeFn,
-  graph: Graph,
-  layout: Layout,
-  style: GraphStyle
-): VizObj {
-  const viz: VizObj = {
```

Comment:
Good that you got rid of this. This functionality really feels like it belongs in a class 👍